Client/game_sa/CModelInfoSA.cpp

```diff
@@ -483,21 +521,278 @@ BOOL CModelInfoSA::DoIsLoaded()
+    if (!MapContains(ms_ModelDefaultFlagsMap, m_dwModelID))
+        MapSet(ms_ModelDefaultFlagsMap, m_dwModelID, m_pInterface->usFlags);
+
+    // Don't change bIsColLoaded flag
+    usFlags &= 0xFF7F;                                  // Disable flag in input
+    usFlags |= m_pInterface->usFlags & 0x80;            // Apply current bIsColLoaded flag
+
+    m_pInterface->usFlags = usFlags;
+}
+
+void CModelInfoSA::SetIdeFlags(unsigned int uiFlags)
+{
+    m_pInterface = ppModelInfo[m_dwModelID];
+    if (!m_pInterface)
+        return;
+
+    // Save default value if not done yet
+    if (!MapContains(ms_ModelDefaultFlagsMap, m_dwModelID))
+        MapSet(ms_ModelDefaultFlagsMap, m_dwModelID, m_pInterface->usFlags);
+
+    tIdeFlags ideFlags;
+    ideFlags.uiFlags = uiFlags;
+
+    // Default value is 0xC0 (bIsColLoaded + bIsBackfaceCulled)
+    // But bIsColLoaded should not be changed
+    m_pInterface->usFlags &= 0x80;            // Reset all flags except bIsColLoaded
+    m_pInterface->bIsBackfaceCulled = true;
+
+    // setBaseModelInfoFlags
+    if (ideFlags.bDrawLast)
     {
-        push    ModelID
-        call    dwFunc
-        add     esp, 4
-        mov     bFlags, al
+        m_pInterface->bAlphaTransparency = true;
+        m_pInterface->bAdditiveRender = true;
+    }
+
+    m_pInterface->bAdditiveRender = ideFlags.bAddictive;
+    m_pInterface->bDontWriteZBuffer = ideFlags.bNoZBufferWrite;
+    m_pInterface->bDontCastShadowsOn = ideFlags.bDontReceiveShadows;
+    m_pInterface->bIsBackfaceCulled = !ideFlags.bDisableBackfaceCilling;
+
+    switch (GetModelType())
+    {
+        case eModelInfoType::ATOMIC:
+        case eModelInfoType::TIME:
+        {
+            // SetAtomicModelInfoFlags
+            m_pInterface->bIsRoad = ideFlags.bIsRoad;
+
+            if (ideFlags.bIsGlassType1)
+                m_pInterface->eSpecialModelType = eModelSpecialType::GLASS_1;
+
+            if (ideFlags.bIsGlassType2)
+                m_pInterface->eSpecialModelType = eModelSpecialType::GLASS_2;
+
+            if (ideFlags.bIsGarageDoor)
+                m_pInterface->eSpecialModelType = eModelSpecialType::GARAGE_DOOR;
+
+            if (ideFlags.bIsTree)
+                m_pInterface->eSpecialModelType = eModelSpecialType::TREE;
+
+            if (ideFlags.bIsPalm)
+                m_pInterface->eSpecialModelType = eModelSpecialType::PALM;
+
+            m_pInterface->bDontCollideWithFlyer = ideFlags.bDontCollideWithFlyer;
+
+            if (ideFlags.bIsTag)
+                m_pInterface->eSpecialModelType = eModelSpecialType::TAG;
+
+            if (ideFlags.bIsBreakableStatue)
+                m_pInterface->eSpecialModelType = eModelSpecialType::BREAKABLE_STATUE;
+
+            if (ideFlags.bFlag24)
+                m_pInterface->eSpecialModelType = eModelSpecialType::UNKNOW_1;
+
+            // Added
+            if (ideFlags.bFlag20)
+                m_pInterface->eSpecialModelType = eModelSpecialType::CRANE;
```

Comment:
This is almost original GTA SA behavior. Game reads flags in the same way

Reply:
I dont know.. Its really weird, because `m_pInterface->eSpecialModelType` will always be overwritten by the last `true` if, thus and `else if` ends up doing the same.

Reply:
if i will use `else if` `m_pInterface->eSpecialModelType` will be overwritten with first `if`

Reply:
Sounds good, thx

Reply:
You're welcome!